CPT331.Core/ObjectModel/Coordinate.cs

```diff
@@ -9,10 +9,8 @@ namespace CPT331.Core.ObjectModel
 {
 	public class Coordinate
 	{
-		public Coordinate(double latitude, double longitude)
+		private Coordinate(double latitude, double longitude)
```

Comment:
This type should only be able to be created with correct values - moved the constructor to private and added a factory method (see below).